src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -690,51 +699,55 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
+    }
+
     const isInputZero = equalWorklet(SwapInputController.inputValues.value.inputAmount, 0);
     const isOutputZero = equalWorklet(SwapInputController.inputValues.value.outputAmount, 0);
 
     const userHasNotEnteredAmount = SwapInputController.inputMethod.value !== 'slider' && isInputZero && isOutputZero;
-
     const userHasNotMovedSlider = SwapInputController.inputMethod.value === 'slider' && SwapInputController.percentageToSwap.value === 0;
 
     if (userHasNotEnteredAmount || userHasNotMovedSlider) {
       return { label: enterAmount, disabled: true, opacity: 1, type: 'hold' };
     }
 
-    if (
-      [SwapWarningType.no_quote_available, SwapWarningType.no_route_found, SwapWarningType.insufficient_liquidity].includes(
-        SwapWarning.swapWarning.value.type
-      )
-    ) {
-      return { icon: '􀕹', label: review, disabled: true, type: 'hold' };
-    }
-
-    const sellAsset = internalSelectedInputAsset.value;
-    const enoughFundsForSwap =
-      sellAsset && lessThanOrEqualToWorklet(SwapInputController.inputValues.value.inputAmount, sellAsset.maxSwappableAmount);
-
-    if (!enoughFundsForSwap) {
-      return { label: insufficientFunds, disabled: true, type: 'hold' };
-    }
+    const holdLabel = swapInfo.value.isBridging ? holdToBridge : holdToSwap;
+    const reviewLabel = SwapSettings.degenMode.value ? holdLabel : review;
 
     const isQuoteError = quote.value && 'error' in quote.value;
     const isLoadingGas = !isQuoteError && hasEnoughFundsForGas.value === undefined;
     const isReviewSheetOpen = configProgress.value === NavigationSteps.SHOW_REVIEW || SwapSettings.degenMode.value;
 
-    if ((isFetching.value || isLoadingGas) && !isQuoteError) {
-      const disabled = (isReviewSheetOpen && (isFetching.value || isLoadingGas)) || !quote.value;
+    const isStale =
+      !!isQuoteStale.value &&
+      (SwapInputController.inputMethod.value !== 'slider' || sliderPressProgress.value === SLIDER_COLLAPSED_HEIGHT / SLIDER_HEIGHT);
+
+    if ((isFetching.value || isLoadingGas || isStale) && !isQuoteError) {
+      const disabled = (isReviewSheetOpen && (isFetching.value || isLoadingGas || isStale)) || !quote.value;
       const buttonType = isReviewSheetOpen ? 'hold' : 'tap';
       return { label: fetchingPrices, disabled, type: buttonType };
     }
 
-    const reviewLabel = SwapSettings.degenMode.value ? holdToSwap : review;
+    const quoteUnavailable = [
+      SwapWarningType.no_quote_available,
+      SwapWarningType.no_route_found,
+      SwapWarningType.insufficient_liquidity,
+    ].includes(SwapWarning.swapWarning.value.type);
 
-    if (isQuoteError) {
+    if (quoteUnavailable || isQuoteError) {
       const icon = isReviewSheetOpen ? undefined : '􀕹';
       return { icon, label: isReviewSheetOpen ? quoteError : reviewLabel, disabled: true, type: 'hold' };
     }
 
-    if (!hasEnoughFundsForGas.value) {
+    if (hasEnoughFundsForGas.value === false) {
```

Comment:
curious if this was because of the `undefined` initialization or another reason

Reply:
Yeah it was, `undefined` represents that we don't yet know if the user has insufficient funds for gas — only `false` should trip this error case